Nice, thank you @Mustaphayinka.

For acceptance testing, would you be able to confirm that the new links bring us to the desired targets? I would usually ask for a screenshot for docs related updates but I realize that might not be the most appropriate for checking link targets. As an aside - I wonder if there is a GitHub actions workflow we can set up to check for broken links in the docs and make sure they all resolve. Maybe we can create a new issue to add that check!

I see the CI build is currently failing, but due to line too long errors.

We can try adding a pylint disable comment (on the tail / closing end of the docstring triple quotes should work I believe).

```
************* Module smart_control.dataset.partition
smart_control/dataset/partition.py:87:0: C0301: Line too long (113/80) (line-too-long)
smart_control/dataset/partition.py:149:0: C0301: Line too long (125/80) (line-too-long)
smart_control/dataset/partition.py:151:0: C0301: Line too long (109/80) (line-too-long)
smart_control/dataset/partition.py:170:0: C0301: Line too long (135/80) (line-too-long)
smart_control/dataset/partition.py:172:0: C0301: Line too long (120/80) (line-too-long)
smart_control/dataset/partition.py:193:0: C0301: Line too long (135/80) (line-too-long)
smart_control/dataset/partition.py:194:0: C0301: Line too long (189/80) (line-too-long)
smart_control/dataset/partition.py:215:0: C0301: Line too long (125/80) (line-too-long)
smart_control/dataset/partition.py:216:0: C0301: Line too long (179/80) (line-too-long)
```

--- a/smart_control/dataset/dataset.py
+++ b/smart_control/dataset/dataset.py
@@ -68,12 +68,15 @@ class BuildingDataset:
     return os.path.join(DATA_DIR, self.zip_filename)
 
   @property
+  # pylint: disable=line-too-long
   def building_dirpath(self):
     """The local directory containing the building's dataset, after it has been
     extracted from the local zip file.
     """
+    # pylint: enable=line-too-long
     return os.path.join(DATA_DIR, self.dataset_id)
 
+  # pylint: disable=line-too-long
   def download(self, timeout=60):
     """Downloads the building's dataset from Google Cloud Storage.
 
@@ -83,6 +86,7 @@ class BuildingDataset:
 
     Download speed is fairly quick, but unzipping takes a few moments.
     """
+    # pylint: enable=line-too-long
     if os.path.isdir(self.building_dirpath):
       print("Using previously-downloaded data...")
       print(os.path.abspath(self.building_dirpath))
@@ -110,6 +114,7 @@ class BuildingDataset:
     return os.path.join(self.tabular_dirpath, "floorplan.npy")
 
   @cached_property
+  # pylint: disable=line-too-long
   def floorplan(self) -> np.ndarray:
     """The building's floorplan, as a numpy array.
 
@@ -122,6 +127,7 @@ class BuildingDataset:
     Use the [`display_floorplan`][smart_control.dataset.dataset.BuildingDataset.display_floorplan]
       method to view an image of the floorplan.
     """
+    # pylint: enable=line-too-long
     return np.load(self.floorplan_filepath)
 
   @property
@@ -130,6 +136,7 @@ class BuildingDataset:
     floorplan_image_filename = f"{self.dataset_id}_floorplan.png"
     return os.path.join(DOCS_DIR, "assets", "images", floorplan_image_filename)
 
+  # pylint: disable=line-too-long
   def display_floorplan(
       self,
       cmap="binary",
@@ -151,6 +158,7 @@ class BuildingDataset:
       image_filepath (str): An optional custom filepath to use when saving the
         image. Only applies if `save=True`. By default, saves to the [`floorplan_image_filepath`][smart_control.dataset.dataset.BuildingDataset.floorplan_image_filepath]
     """
+    # pylint: enable=line-too-long
     plt.imshow(self.floorplan, interpolation="nearest", cmap=cmap)
     if show:
       plt.show()

--- a/smart_control/dataset/partition.py
+++ b/smart_control/dataset/partition.py
@@ -77,12 +77,12 @@ class BuildingDatasetPartition:
         See corresponding documentation below for more information about each.
     """
     return np.load(self.data_filepath)
-
   @property
   def metadata_filepath(self):
     return os.path.join(self.partition_dirpath, "metadata.pickle")
 
   @cached_property
+  # pylint: disable=line-too-long
   def metadata(self) -> dict:
     """Metadata describing the partition [`data`][smart_control.dataset.partition.BuildingDatasetPartition.data].
 
@@ -100,6 +100,7 @@ class BuildingDatasetPartition:
         Each of these keys has a corresponding public method for convenience.
         See corresponding documentation below for more information about each.
     """
+    # pylint: enable=line-too-long
     metadata = pickle.load(open(self.metadata_filepath, "rb"))
     # renaming keys:
     metadata = {
@@ -136,12 +137,12 @@ class BuildingDatasetPartition:
   def reward_info_value_matrix(self) -> np.ndarray:
     """Time series reward information data."""
     return self.data["reward_info_value_matrix"]
-
   #
   # METADATA PROPERTIES
   #
 
   @cached_property
+  # pylint: disable=line-too-long
   def action_ids_map(self) -> dict:
     """A mapping of unique action identifiers.
 
@@ -152,17 +153,19 @@ class BuildingDatasetPartition:
 
         For example:
 
-        ```py
+```py
           {
             '12945159110931775488@supply_air_temperature_setpoint': 0,
             '13761436543392677888@supply_water_temperature_setpoint': 1,
             '14409954889734029312@supply_air_temperature_setpoint': 2
           }
-        ```
+```
     """
+    # pylint: enable=line-too-long
     return self.metadata["action_ids_map"]
 
   @cached_property
+  # pylint: disable=line-too-long
   def observation_ids_map(self) -> dict:
     """A mapping of unique observation identifiers.
 
@@ -173,17 +176,19 @@ class BuildingDatasetPartition:
 
         For example:
 
-        ```py
+```py
           {
             '202194278473007104@building_air_static_pressure_setpoint', 0,
             ...
             '2640423556868160@zone_air_temperature_sensor': 1197
           }
-        ```
+```
     """
+    # pylint: enable=line-too-long
     return self.metadata["observation_ids_map"]
 
   @cached_property
+  # pylint: disable=line-too-long
   def reward_info_ids_map(self) -> dict:
     """A mapping of unique reward info identifiers.
 
@@ -195,17 +200,19 @@ class BuildingDatasetPartition:
 
         For example:
 
-        ```py
+```py
           {
             'rooms/9028552126@heating_setpoint_temperature': 0
             ...
             '14409954889734029312@air_conditioning_electrical_energy_rate': 3251
           }
-        ```
+```
     """
+    # pylint: enable=line-too-long
     return self.metadata["reward_info_ids_map"]
 
   @cached_property
+  # pylint: disable=line-too-long
   def reward_ids_map(self) -> dict:
     """A mapping of unique reward identifiers.
 
@@ -215,6 +222,7 @@ class BuildingDatasetPartition:
       A dictionary where the keys are the [`reward_ids`][smart_control.dataset.partition.BuildingDatasetPartition.reward_ids]
         and the values are unique integers referencing column indices in the [`reward_value_matrix`][smart_control.dataset.partition.BuildingDatasetPartition.reward_value_matrix].
     """
+    # pylint: enable=line-too-long
     return {
         "agent_reward_value": 0,
         "productivity_reward": 1,
@@ -235,6 +243,7 @@ class BuildingDatasetPartition:
         "normalized_carbon_emission": 16,
     }
 
+    
   @cached_property
   def action_ids(self) -> list[str]:
     """A list of unique action identifiers.

--- a/smart_control/simulator/building_radiation_utils.py
+++ b/smart_control/simulator/building_radiation_utils.py
@@ -76,13 +76,19 @@ def calculate_a_tilde_inv(epsilon: np.ndarray, F: np.ndarray) -> np.ndarray:
   return np.linalg.solve(a_tilde, np.eye(n))
 
 
+# pylint: disable=line-too-long
+# pylint: disable=line-too-long
+# pylint: disable=line-too-long
+# pylint: disable=line-too-long
+# pylint: disable=line-too-long
+# pylint: disable=line-too-long
 def calculate_ifa_inv(F: np.ndarray, A_inv: np.ndarray) -> np.ndarray:
   r"""
   Calculates the $IFA_{inv}$ matrix.
 
   $$IFA_{inv} = (I - F) \tilde{A}^{-1}$$
 
-  See [`net_radiative_heatflux_function_of_T`][smart_control.simulator.building_radiation_utils.net_radiative_heatflux_function_of_T] for more details.
+  See [`net_radiative_heatflux_function_of_T`][smart_control.simulator.building_radiation_utils.net_radiative_heatflux_function_of_t] for more details.
 
   Args:
       F (np.ndarray): The view factor matrix.
@@ -91,6 +97,12 @@ def calculate_ifa_inv(F: np.ndarray, A_inv: np.ndarray) -> np.ndarray:
   Returns:
       IFA_inv : The IFA inverse matrix.
   """
+  # pylint: enable=line-too-long
+  # pylint: enable=line-too-long
+  # pylint: enable=line-too-long
+  # pylint: enable=line-too-long
+  # pylint: enable=line-too-long
+  # pylint: enable=line-too-long
 
   n = F.shape[0]
 

